include/gul14/to_number.h

```diff
@@ -563,6 +565,33 @@ constexpr inline optional<NumberType> to_number(gul14::string_view str) noexcept
     return detail::to_unsigned_float<NumberType>(str);
 }
 
+template<typename NumberType, std::enable_if_t<std::is_same<NumberType, bool>::value, int> = 0>
+constexpr inline optional<NumberType> to_number(gul14::string_view str) noexcept
+{
+    size_t pos{};
+    bool value{};
+
+    for (; pos < str.length(); ++pos)
+    {
+        if (not std::isdigit(str[pos]))
+            break;
+
+        if (str[pos] != '0')
```

Comment:
This treats `2` as true, which is inconsistent with the behavior for other integral types. For instance, `to_number<std::uint8_t>(256)` is `nullopt` because the number exceeds the range of the integer type.